File: recomp-mobile/app/src/main/java/com/refactor/app/api/RicoReplySanitizer.kt
package com.refactor.app.api

/** Removes temporary Rico meal-logging debug markup (e.g. `[DIAG today=…]` blocks). */
fun stripRicoDiagnosticMarkup(text: String): String =
    text.replace(Regex("""\[DIAG[\s\S]*?\]""", RegexOption.IGNORE_CASE), "")
        .replace(Regex("\n{3,}"), "\n\n")
        .trim()

File: src/lib/rico-reply-sanitizer.ts
/** Strip temporary Rico meal-logging debug markup from chat replies. */
export function stripRicoDiagnosticMarkup(text: string): string {
  return text
    .replace(/\[DIAG[\s\S]*?\]/gi, "")
    .replace(/\n{3,}/g, "\n\n")
    .trim();
}
